src/map.rs

```diff
@@ -1348,6 +1348,40 @@ where
         self.put(key, value, false, guard)
     }
 
+    /// Tries to insert a key-value pair into the map.
+    ///
+    /// If the map did not have this key present, `Ok(())` is returned and the
```

Comment:
This is _very_ nitty, but I prefer the active, present voice here (and in general):
> If the map does not contain the key, the key-value pair is inserted and this method returns `Ok`.
>
> If the map does contain the key, the map is left unchanged and this method returns `Err`.

The "map is left unchanged" also covers the `==` case implicitly.